kie-osgi/kie-karaf-itests/src/test/java/org/kie/karaf/itest/planner/PlannerCloudBalanceIntegrationTest.java

```diff
@@ -0,0 +1,218 @@
+import org.ops4j.pax.exam.spi.reactors.PerClass;
+import org.optaplanner.core.api.score.buildin.hardsoft.HardSoftScore;
+import org.optaplanner.core.api.solver.Solver;
+import org.optaplanner.core.api.solver.SolverFactory;
+import org.optaplanner.core.config.SolverConfigContext;
+import org.optaplanner.core.config.constructionheuristic.ConstructionHeuristicPhaseConfig;
+import org.optaplanner.core.config.constructionheuristic.ConstructionHeuristicType;
+import org.optaplanner.core.config.localsearch.LocalSearchPhaseConfig;
+import org.optaplanner.core.config.phase.PhaseConfig;
+import org.optaplanner.core.config.score.definition.ScoreDefinitionType;
+import org.optaplanner.core.config.score.director.ScoreDirectorFactoryConfig;
+import org.optaplanner.core.config.solver.SolverConfig;
+import org.optaplanner.core.config.solver.termination.TerminationConfig;
+
+import static org.ops4j.pax.exam.karaf.options.KarafDistributionOption.configureConsole;
+import static org.ops4j.pax.exam.karaf.options.KarafDistributionOption.debugConfiguration;
+import static org.ops4j.pax.exam.karaf.options.KarafDistributionOption.logLevel;
+
+@RunWith(PaxExam.class)
+@ExamReactorStrategy(PerClass.class)
+public class PlannerCloudBalanceIntegrationTest extends AbstractKarafIntegrationTest {
+
+    private static final String CLOUD_BALANCE_INCREMENTAL_CONFIG_FILE = "cloudBalanceIncrementalConfig.xml";
+    private static final String CLOUD_BALANCE_DROOLS_CONFIG_FILE = "cloudBalanceDroolsConfig.xml";
+
+    @Test(expected=IllegalStateException.class)
+    public void invalidSolutionPlannerTest() {
+        solveSolution(cloudBalanceGeneratorForFuse(0, 0, 0));
+    }
+
+    @Test
+    public void plannerCloudBalanceTest() {
+        solveSolution(cloudBalanceGeneratorForFuse(5, 10, 0));
+    }
+
+    @Test
+    @Ignore("RHPAM-1554")
+    public void plannerSolverFactoryTest() throws IOException {
+        InputStream stream = this.getClass().getResourceAsStream(CLOUD_BALANCE_INCREMENTAL_CONFIG_FILE);
+        SolverFactory solverFactory = SolverFactory
+                .createFromXmlInputStream(stream, PlannerCloudBalanceIntegrationTest.class.getClassLoader());
+        stream.close();
+
+        CloudBalance cloudBalance = cloudBalanceGeneratorForFuse(10, 20, 0);
+        Assert.assertEquals(cloudBalance.getScore().getHardScore(), Integer.MIN_VALUE);
+        Assert.assertEquals(cloudBalance.getScore().getSoftScore(), Integer.MIN_VALUE);
+
+        Solver solver = solverFactory.getSolverConfig().buildSolver(new SolverConfigContext());
+        solver.solve(cloudBalance);
+
+        CloudBalance solution = (CloudBalance) solver.getBestSolution();
+
+        Assert.assertNotEquals(solution.getScore().getHardScore(), Integer.MIN_VALUE);
+        Assert.assertNotEquals(solution.getScore().getSoftScore(), Integer.MIN_VALUE);
+    }
+
+    @Test
+    @Ignore("RHPAM-1554")
+    public void plannerSolverFactoryDroolsTest() throws IOException {
```

Comment:
This one and the previous test can be refactored into a private method taking a location of the config file on classpath as a parameter.

Reply:
@rsynek Thank you for your comment. Now it uses one shared private method.